msm/src/serialization/constraints.rs

```diff
@@ -185,3 +185,71 @@ impl<F: PrimeField> InterpreterEnv<F> for Env<F> {
         }))
     }
 }
+
+impl<Fp: PrimeField> Env<Fp> {
+    #[allow(dead_code)]
+    // FIXME: not mut
+    fn constrain_lookups(&mut self) -> Vec<E<Fp>> {
+        assert_eq!(self.rangecheck4_lookups.len(), 20);
```

Comment:
This means each row adds always 20 rangecheck4 lookups, and 17 rangecheck15's? Probably this is for the limbs checks?

Reply:
Yes